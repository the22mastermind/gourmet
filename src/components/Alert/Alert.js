import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import { Snackbar } from 'react-native-paper';
import { AlertContext } from '../../context/AlertProvider';

const styles = StyleSheet.create({
  error: {
    backgroundColor: '#f44336',
  },
  info: {
    backgroundColor: '#2196f3',
  },
  success: {
    backgroundColor: '#4caf50',
  },
});

const Alert = ({ testID, type, message }) => {
  const [visible, setVisible] = React.useState(true);
  const { clearAlert } = useContext(AlertContext);

  const onDismissSnackBar = async () => {
    setVisible(false);
    await clearAlert();
  };

  return (
    <>
      <Snackbar
        testID={testID}
        accessibilityLabel={testID}
        visible={visible}
        duration={5000}
        onDismiss={onDismissSnackBar}
        style={type === 'error' ? styles.error : type === 'info' ? styles.info : styles.success}
        action={{
          label: 'Dismiss',
          onPress: () => onDismissSnackBar(),
        }}>
        {message}
      </Snackbar>
    </>
  );
};

Alert.propTypes = {
  testID: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};

export default Alert;
