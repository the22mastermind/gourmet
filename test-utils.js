import React from 'react';
import { render } from '@testing-library/react-native';
import PaperProvider from 'react-native-paper/lib/commonjs/core/Provider';

const AllTheProviders = ({ children }) => (
  <PaperProvider>{children}</PaperProvider>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
