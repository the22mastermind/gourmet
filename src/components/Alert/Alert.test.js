import React from 'react';
import { render, fireEvent, waitForElementToBeRemoved } from 'test-utils';
import Alert from './Alert';

describe('<Alert /> component', () => {
  it('renders an error alert', () => {
    const screen = render(
      <Alert
        testID="snackbar-alert"
        type="error"
        message="This is an error message"
      />);
  
    expect(screen.getByText('This is an error message')).toBeTruthy();
  });

  it('renders an info alert', () => {
    const screen = render(
      <Alert
        testID="snackbar-alert"
        type="info"
        message="This is an info message"
      />);
  
    expect(screen.getByText('This is an info message')).toBeTruthy();
  });

  it('renders a success alert', () => {
    const screen = render(
      <Alert
        testID="snackbar-alert"
        type="success"
        message="This is a success message"
      />);
  
    expect(screen.getByText('This is a success message')).toBeTruthy();
  });
  
  it('should dismiss the alert correctly', async () => {
    const screen = render(
      <Alert
        testID="snackbar-alert"
        type="success"
        message="This is a success message"
      />);
    
    fireEvent.press(screen.getByRole('button'));
  
    await waitForElementToBeRemoved(() => screen.queryByText('This is a success message'));
  });
  
  it('matches <Alert /> component snapshot', async () => {
    const screen = render(
      <Alert
        testID="snackbar-alert"
        type="error"
        message="This is an error message"
      />);
  
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
