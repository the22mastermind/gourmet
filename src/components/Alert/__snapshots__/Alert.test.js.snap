// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> component matches <Alert /> component snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityLabel="snackbar-alert"
      accessibilityLiveRegion="polite"
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f44336",
          "borderRadius": 4,
          "elevation": 6,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 8,
          "opacity": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 6,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
        }
      }
      testID="snackbar-alert"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "flex": 1,
                "flexWrap": "wrap",
                "marginLeft": 16,
                "marginVertical": 14,
              },
              Object {
                "color": "#ffffff",
                "marginRight": 0,
              },
            ],
          ]
        }
      >
        This is an error message
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "marginHorizontal": 8,
            "marginVertical": 6,
            "minWidth": "auto",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#f7f8f8",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Dismiss
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
